identity/app/form/PrivacyMapping.scala

```diff
@@ -108,17 +110,41 @@ case class PrivacyFormData(
   }
 }
 
-object PrivacyFormData {
+object PrivacyFormData extends SafeLogging {
   /**
     * Converts User DO from IDAPI to form processing DTO PrivacyFromData
     *
     * @param userDO Identity User domain model from IDAPI defiend in identity-model library
     * @return form processing DTO PrivacyFromData
     */
-  def apply(userDO: User): PrivacyFormData =
+  def apply(userDO: User): PrivacyFormData = {
     PrivacyFormData(
       receiveGnmMarketing = userDO.statusFields.receiveGnmMarketing,
       receive3rdPartyMarketing = userDO.statusFields.receive3rdPartyMarketing,
       allowThirdPartyProfiling = userDO.statusFields.allowThirdPartyProfiling,
-      consents = if (userDO.consents.isEmpty) defaultConsents else userDO.consents)
+      consents = if (userDO.consents.isEmpty) defaultConsents else onlyValidConsents(userDO)
+    )
+  }
+
+  /**
+    * FIXME: Once GDPR goes live, clean Mongo DB of old consents, and remove this method.
+    *
+    * Filter out any invalid consents that are still lingering in Mongo DB.
+    *
+    * For example, if consent id is renamed, then some users might still have the old consent.
+    *
+    * @param userDO Identity User domain model from IDAPI defiend that might contain some old invalid consents
+    * @return list of valid consents
+    */
+  private def onlyValidConsents(userDO: User): List[Consent] = {
+    def consentExistsInModel(consent:Consent): Boolean =
+      Try(Consent.wording(consent.id, consent.version)).isSuccess
```

Comment:
it feels like `Consent.wording` should probably have been returning an Option

Reply:
We will address it in a separate PR.